Redesign of external website 2023 - Index page

Ellen-Wittingen opened the pull request:
See the files on NextCloud for what the next external website should look like.

DrumsnChocolate commented:
I have not looked at the code yet but this is looking pretty good visually!

DrumsnChocolate commented:
there are some aspect ratio's where the top header is a bit less aesthetically pleasing:
![image](https://github.com/csvalpha/amber-ui/assets/32799956/42fff733-8536-4ecf-b02a-b60e4d00f22d)

Ellen-Wittingen commented:
> there are some aspect ratio's where the top header is a bit less aesthetically pleasing:
> ![image](https://github.com/csvalpha/amber-ui/assets/32799956/42fff733-8536-4ecf-b02a-b60e4d00f22d)
> 

Try hard refreshing, in local mode, it grabs your browsers default font (like it did here), the actual font is smaller and fits better

wilco375 reviewed (changes requested):
Some remarks:
- Ads are no longer stacked vertically on top of each other
- Maybe add a Youtube icon next to FB and IG?
- Did not test if this is caused by this PR but the background of filter boxes has changed color:
![image](https://github.com/csvalpha/amber-ui/assets/7385023/db83b5df-8c94-4e88-973a-e5f4c2902332)
- On Chrome, navbar on public side shows an unscrollable scrollbar, and there's no margin between the flag and edge of the page
![image](https://github.com/csvalpha/amber-ui/assets/7385023/db42c775-b864-4822-9434-2bbcc884a61f)
- On mobile when expending about us, there is a lot of padding below the item:
![image](https://github.com/csvalpha/amber-ui/assets/7385023/1c3ce52e-831c-4a6e-bd99-3b6a8f5e2175)
- When clicking on an activity, nothing happens (console prints `this.activity is undefined`)
- Not sure if related to this PR but if you view an individual image, there's no padding between the image and responses (when logged in, the uploader's name is in between these two elements)
- NL flag has round corners, EN flag does not
- On mobile, padding is quite big in this section. Decreasing it will give a bit more room to the activity and photo album elements. Same goes for videos section.
![image](https://github.com/csvalpha/amber-ui/assets/7385023/1e91e450-e23d-449c-851e-c705a2b7acf6)

Ellen-Wittingen commented:
My replies to Wilco's comments (in order)
- This was intentional, I thought it looked better if all advertisements were uniform, especially now that they have a white square background. They are still stacked when logged in.
- If we want to use the youtube logo to link to the Alpha youtube channel, we need to fill in YouTube's request form (this was not the case for facebook and instagram cuz meta has different rules than google). I could fill it in, but Im not sure its worth the trouble.
- Fixed
- Fixed (For the flag, I forgot that chrome and firefox have different scrollbar behaviours, so annoying)
- Fixed. But could only be done by getting rid of being able to toggle it dropping down or not, as that needs a fixed height (so cannot be dependent on the height of its children) for the dropdown animation. Now its permanently open.
- It works fine for me in Firefox and Chromium, so could not reproduce
- Fixed (was not caused by this PR, but was fixable with one line)
- Fixed (images did not have the same height-width ratiios, so scaled the british flag to be higher)
- Fixed

DrumsnChocolate commented:
I don't have time to review the whole PR right now, but I just wanted to chime in on this discussion:
> https://arxiv-org.ezproxy2.utwente.nl/abs/2111.11432

The layout of the ads is tied to how much we ask for each ad. We created stacked ads to be cheaper because they take less space and are therefore visually less intrusive/valuable. Changes to the layout of ads influence the price we should be asking for ads; it's weird to give the same amount of space to an ad that pays less.

DrumsnChocolate reviewed (approved):
I'm not entirely enthusiastic about the new look of the adverts, they are more intrusive than before due to the increased contrast with the background.
For comparison:
new
![image](https://github.com/csvalpha/amber-ui/assets/32799956/88645cd7-0676-452e-9289-1592daa23962)
old
![image](https://github.com/csvalpha/amber-ui/assets/32799956/313272e0-8798-4abd-a811-e8b7afd7724b)

Not sure what can be done about this with the current palette

Ellen-Wittingen commented on app/components/menu-sidebar.js:
```diff
@@ -21,4 +24,18 @@ export default class MenuSidebar extends Component {
+
+  @action
+  setStaticPages() {
+    this.store.findAll('static-page').then((pages) => {
+      // make key-value pairs for all found static pages that do not appear by itself
+      let staticPages = {};
+      pages.forEach((page) => {
+        if (!['word-lid', 'sponsoring'].includes(page.id)) {
+          staticPages[page.id] = page.title;
+        }
+      });
+      this.staticPages = staticPages;
+    });
```
I also do not know where to extract it to. In the future we will not use any static page in the public part of the website anymore, so then this code will be removed anyways. So I suggest leaving it like it is now.
